ceph/utils.py

```diff
@@ -705,46 +705,66 @@ def get_disk_info(node):
     return disks
 
 
-def get_nodes_by_id(cluster, node_names):
+def get_node_by_id(cluster, node_name):
     """
-    Get node object(s) by Node Id
+    Fetch node using provided substring of node name
 
-    Returns all nodes object list if node_names is empty,
-    else node list which matched node ID(eg., 'node1')
+    As per the naming convention used at VM creation, where each node.shortname
+    is framed with hyphen("-") separated string as below,
+        "ceph-<name>-node1-<roles>"
+
+    In this method we use hyphen("-") appended to node_name string to fetch exact node.
+    for example,
+        "node1" ----> node1-
+
+    But however if node_name doesn't follow naming convention as mentioned in
+    inventory, the first searched node will be returned
+    for example,
+        "node" ----> it may return any node which matched first, like node11.
+
+    If this cluster has no nodes with this shortname and
+    node_name did not match any node.shortname, return None.
 
     Args:
         cluster: ceph object
-        node_names: node name list (eg., ['node1'])
-
+        node_name: node1        # try to use node<Id>
     Returns:
-        node_list: list nodes
+        node object
     """
-    nodes = node_names if isinstance(node_names, list) else [node_names]
+    if not isinstance(node_name, str):
+        return None
 
-    # Empty list wll pick all cluster nodes
-    if not nodes:
-        return cluster.get_nodes()
-    else:
-        return [
-            node
-            for node in cluster.get_nodes()
-            for name in nodes
-            if name in node.shortname
-        ]
+    # Appending "-" as per naming convention used at instance creation
+    node_name += "-"
 
+    for node in cluster.get_nodes():
+        if node_name in node.shortname:
+            return node
 
-def get_node_by_id(cluster, node_name):
+
+def get_nodes_by_id(cluster, node_names):
```

Comment:
IMO, this method name could be `get_nodes_by_ids`. As the purpose is to return a list of nodes matching the list of given ids. 

If the intent is that then we can avoid the check of `node_names` being an empty list

Reply:
Done. updated.